web/src/App.tsx

```diff
@@ -0,0 +1,81 @@
+import { useAnalytics } from "@/analytics";
+import { Nav } from "@/components/navBar/Nav";
+import { AnalyticsDashboard } from "@/pages/analytics/AnalyticsDashboard";
+import { NotificationsMap } from "@/pages/dashboards/index";
+import { Box, Center, Flex, Spinner } from "@chakra-ui/react";
+import { createContext, Suspense } from "react";
+import { Navigate, Route, Routes } from "react-router-dom";
+import { useRecoilValue } from "recoil";
+import { Footer } from "./components/Footer";
+import {
+  CityListHookReturnType,
+  useUpdateCityList,
+} from "./data/Hooks/selectCityHook";
+import { connectionConfig } from "./data/recoil";
+import { Overview } from "./pages/configure/Overview";
+
+export let UserContext: React.Context<CityListHookReturnType>;
+
+function App() {
+  const loadingFallback = (
+    <Center height="100vh">
+      <Spinner
+        size="xl"
+        speed="0.85s"
+        thickness="3px"
+        emptyColor="gray.200"
+        color="blue.500"
+      />
+    </Center>
+  );
+
+  return (
+    <Suspense fallback={loadingFallback}>
+      <RoutesContainer />
+    </Suspense>
+  );
+}
+
+const RoutesContainer = () => {
+  const config = useRecoilValue(connectionConfig);
+  const CityListHook = useUpdateCityList(config);
+  UserContext = createContext(CityListHook);
+
+  const Analytics = () => {
+    useAnalytics();
+    return <></>;
+  };
+  return (
+    <>
+      <Analytics />
+      <Flex height="100vh" width="100vw" direction="column" overflowY={"auto"}>
```

Comment:
we usually don't use `vh` or `vw` but rather percentage

Reply:
% only works if there is atleast 1 fixed value on the parents elements

The parent component of this do not have fixed value and are inline. giving % will not work

This is our top most block can consider it as body. rest all component can now have relative values like % because this value is now fixed